Sources/Internal/VisibleItemsProvider.swift

```diff
@@ -564,8 +564,34 @@ final class VisibleItemsProvider {
     if layoutItem.frame.intersects(extendedBounds) {
       numberOfConsecutiveNonIntersectingItems = 0
 
-      // Handle items that actually intersect the visible bounds
+      let month = layoutItem.itemType.month
+
+      // Calculate and the current month frame if it's not cached; it will be used in other
+      // calculations.
+      let monthFrame: CGRect
+      if let cachedMonthFrame = framesForVisibleMonths[month] {
+        monthFrame = cachedMonthFrame
+      } else {
+        let monthOrigin = frameProvider.originOfMonth(containing: layoutItem)
+        monthFrame = frameProvider.frameOfMonth(month, withOrigin: monthOrigin)
+      }
+
+      // Use the calculated month frame to determine content boundaries if needed.
+      determineContentBoundariesIfNeeded(
```

Comment:
The main change here is that we now determine our content boundaries after the `layoutItem.frame.intersects(extendedBounds)` check but not after the `layoutItem.frame.intersects(bounds)` check (as it was before this change). Determining this after the latter check was fragile since it was possible for a user to scroll so fast that they would skip right past a boundary-determining month, sending them into a blank scrolling view with no visual feedback as to what happened.

`extendedBounds` already existed to solve a similar problem, and I should have been using it for this check since the start.